src/BIP44CoinTypeNode.ts

```diff
@@ -147,7 +147,8 @@ export class BIP44CoinTypeNode implements BIP44CoinTypeNodeInterface {
     validateCoinType(coin_type);
     validateCoinTypeNodeDepth(node.depth);
 
-    return new BIP44CoinTypeNode(node, coin_type);
+    // TODO: Make this function not async in a future version.
+    return Promise.resolve(new BIP44CoinTypeNode(node, coin_type));
```

Comment:
Why was this failing lint? This method should already return a promise, no?

Reply:
`async` functions must use `await` or return a `Promise` in the function. This function previously didn't have any async logic, and it should be changed to not return a `Promise`, but doing so would be a breaking change.